app/Helpers/MSGraphAPI/UserByApp.php

```diff
@@ -0,0 +1,216 @@
+            switch($searchProperty){
+                case 'employeeNumber':
+                    $searchProperty = $this->employeeNumberProperty;
+                    $search= $searchProperty && $searchValue ? '&$filter='. $searchProperty . ' eq \'' . $searchValue . '\'' : null;
+                    break;
+                case 'description':
+                    $searchProperty = $this->descriptionProperty;
+                    $search= $searchProperty && $searchValue ? '&$filter='. $searchProperty . ' eq \'' . $searchValue . '\'' : null;
+                    break;
+                case 'email':
+                    $searchProperty = $this->mailProperty; 
+                    $search= $searchProperty && $searchValue ? '&$filter='. $searchProperty . ' eq \'' . $searchValue . '\'' : null;
+                    break;
+                default:
+                    $search = $searchProperty && $searchValue ? '&$search="'. $searchProperty . ':' . $searchValue . '"' : null;
+                    break;
+            }
+            // if search then append to query
+            //$search = $searchProperty && $searchValue ? '&$search="'. $searchProperty . ':' . $searchValue . '"' : null;
+            //$search= $searchProperty && $searchValue ? '&$filter='. $searchProperty . ' eq \'' . $searchValue . '\'' : null;
+            $endpoint = '/users';
+            $url = $endpoint . '?$count=true&$orderBy=displayName' . $topN . $search . ' &$select=id,displayName,surname,givenName,userPrincipalName,mail,businessPhones,onPremisesSamAccountName,employeeId,' . $this->descriptionProperty . ',' . $this->employeeNumberProperty  . ',' . $this->mailProperty ;
+            # https://graph.microsoft.com/v1.0/users?$search="displayName:MyName"&$count=true&$orderBy=displayName&$select=id,displayName,surname,givenName,userPrincipalName,mail,businessPhones,onPremisesSamAccountName,employeeId,extension_e1fd504a055643d5b9d370b8b24b45ee_description,extension_e1fd504a055643d5b9d370b8b24b45ee_employeeNumber,extension_e1fd504a055643d5b9d370b8b24b45ee_mail
+            $request = $graph->createRequest("GET", $url);
+            $request->addHeaders(["ConsistencyLevel"=> "eventual"]);
+            $data=$request->execute();
+            $content = $data->getBody();
+            $users = collect(json_decode(json_encode($content['value']), FALSE));
+            if(empty($users) || count($users) >= 50 ) { 
+                return null; 
+            }
+            // if nextLink then loop
+            while(array_key_exists('@odata.nextLink',$content)){
+                $url = $content['@odata.nextLink'];
+                $request = $graph->createRequest("GET", $url);
+                $request->addHeaders(["ConsistencyLevel"=> "eventual"]);
+                $data=$request->execute();
+                $content = $data->getBody();
+                $newusers = collect(json_decode(json_encode($content['value']), FALSE));
```

Comment:
encode in de decode?

Reply:
de enige manier om snel de geneste data goed te formatten... heb het opgezocht :p

Reply:
ah check